x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java

```diff
@@ -328,13 +350,28 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalString(repositoryName);
             out.writeOptionalString(snapshotName);
             out.writeOptionalString(shrinkIndexName);
+            if (out.getVersion().onOrAfter(Version.V_8_1_0)) {
+                out.writeOptionalLong(indexCreationDate);
+            }
         }
     }
 
     public String getIndex() {
         return index;
     }
 
+    public Long getIndexCreationDate() {
+        return indexCreationDate;
+    }
+
+    public TimeValue getTimeSinceIndexCreation(Supplier<Long> now) {
+        if (indexCreationDate == null) {
+            return null;
+        } else {
+            return TimeValue.timeValueMillis(now.get() - indexCreationDate);
```

Comment:
Ah, one more thing - I think we should guard against negative values as `timeValueMillis` will fail if it encounters them.

Maybe do something along the lines of 

```suggestion
            return TimeValue.timeValueMillis(Math.min(0L, now.get() - indexCreationDate));
```

`getAge` suffers from the same problem.

Reply:
Good catch!

Reply:
Fixed & tested